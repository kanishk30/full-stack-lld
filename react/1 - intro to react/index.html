<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>

<body>
    <div id="root"></div>
</body>

<script>
    const headingElem = document.createElement('h1');
    headingElem.innerText = 'Hello from JS';

    const root = document.getElementById('root');
    root.appendChild(headingElem);

</script>

</html>