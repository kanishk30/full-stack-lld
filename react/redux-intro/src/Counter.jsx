import React from 'react';

import { useSelector, useDispatch } from 'react-redux';
import { increment, decrement } from './redux/counterSlice';
// use the reduc state. update the action

const Counter = () => {
    const count = useSelector((state) => state.counter.value); // currents tate isfetched.
    const dispatch = useDispatch()
    return (
        <div>
            <button onClick={() => dispatch(decrement())}>
                decrement
            </button>
            <h2>Count: {count}</h2>
            <button onClick={() => dispatch(increment())}>
                increment
            </button>

        </div>
    )
}

export default Counter;