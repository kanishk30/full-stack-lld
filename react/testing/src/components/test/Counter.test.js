import { render, screen, fireEvent } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Counter from '../Counter'
/**
 * 1. initial state of the count, it should be 0.
 * 2. increment.
 *      - on increment button click, new count shud be +1.
 * 3. decrement
 *      - on decrement btn click, new count shud be -1.
 */

describe('Counter component testcases', () => {
    test('initial state check', () => {
        // rendering in VDOM
        render(<Counter></Counter>);

        // selection..
        const countText = screen.getByText('Count is 0');
        const decrBtn = screen.getByText('Decrement');
        const incrBtn = screen.getByText('Increment');

        // we are verifying...
        expect(countText).toBeInTheDocument()
        expect(decrBtn).toBeInTheDocument()
        expect(incrBtn).toBeInTheDocument()
    })
    test('incr by 1', () => {
        render(<Counter></Counter>);
        const incrBtn = screen.getByText('Increment');
        fireEvent.click(incrBtn);
        const countTextOne = screen.getByText('Count is 1');
        expect(countTextOne).toBeInTheDocument()
    })
    test('decr by 2', () => {
        render(<Counter></Counter>);
        const decrBtn = screen.getByText('Decrement');
        fireEvent.click(decrBtn);
        fireEvent.click(decrBtn);
        const countTextMinusTwo = screen.getByText('Count is -2');
        expect(countTextMinusTwo).toBeInTheDocument()
    })

    // snapshot...
    test('snapshot for counter', () => {
        const { asFragment } = render(<Counter />)
        expect(asFragment()).toMatchSnapshot()

    })
})