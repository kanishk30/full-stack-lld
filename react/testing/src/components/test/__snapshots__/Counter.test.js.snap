// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component testcases snapshot for counter 1`] = `
<DocumentFragment>
  <h2>
    Counter application
  </h2>
  <button>
     Decrement  
  </button>
  <h4>
    Count is 0
  </h4>
  <button>
     Increment  
  </button>
</DocumentFragment>
`;
